docs/topics/data-analysis/data-analysis-connect-to-db.md

```diff
@@ -1,22 +1,167 @@
+
+## Retrieve and manipulate data
+
+After [establishing a connection to an SQL database](#connect-to-database), you can retrieve and manipulate data in Kotlin Notebook, utilizing the DataFrame library. 
+You can use `readSqlTable` to retrieve data. To manipulate data, you can use methods, such as [`.filter`](https://kotlin.github.io/dataframe/filter.html), [`.groupBy`](https://kotlin.github.io/dataframe/groupby.html), 
+and [`.convert`](https://kotlin.github.io/dataframe/convert.html). 
+
+Let's look at an example of connecting to an IMDB database and retrieving data about movies directed by Quentin Tarantino:
+
+1. **Fetch data from a table:** Use the `readSqlTable` function to retrieve data from the "movies" table, setting `limit` 
+to restrict the query to the first 100 records for efficiency:
+
+```kotlin
+val dfs = DataFrame.readSqlTable(dbConfig, tableName = "movies", limit = 100)
+```
+
+2. **Execute SQL query:**
+Use an SQL query to retrieve a specific dataset related to movies directed by Quentin Tarantino. 
+This query selects movie details and combines genres for each movie:
+
+```kotlin
+val props = Properties()
+props.setProperty("user", USER_NAME)
+props.setProperty("password", PASSWORD)
+
+val TARANTINO_FILMS_SQL_QUERY = """
+    SELECT name, year, rank, GROUP_CONCAT(genre) as "genres"
+    FROM movies JOIN movies_directors ON movie_id = movies.id
+    JOIN directors ON directors.id=director_id LEFT JOIN movies_genres ON movies.id = movies_genres.movie_id
+    WHERE directors.first_name = "Quentin" AND directors.last_name = "Tarantino"
+    GROUP BY name, year, rank
+    ORDER BY year
+    """
+
+var dfTarantinoMovies: DataFrame<*>
+
+DriverManager.getConnection(URL, props).use { connection ->
+   connection.createStatement().use { st ->
+      st.executeQuery(TARANTINO_FILMS_SQL_QUERY).use { rs ->
+         val dfTarantinoFilmsSchema = DataFrame.getSchemaForResultSet(rs, connection)
+         dfTarantinoFilmsSchema.print()
+
+         dfTarantinoMovies = DataFrame.readResultSet(rs, connection)
+         dfTarantinoMovies
+      }
+   }
+}
+```
+
+3. **Manipulate and filter data:**
+After fetching the Tarantino movies dataset, you can further manipulate and filter the data.
+
+```kotlin
+val df = dfTarantinoMovies
+    .fillNA { year }.with { 0 }
+    .convert { year }.toInt()
+    .filter { year > 2000 }
+```
+
+The resulting output is a DataFrame where missing values in the year column are replaced with 0 using the 
+[`.fillNA`](https://kotlin.github.io/dataframe/fill.html#fillna) method, the year column is converted to integer values 
+with the [`.convert`](https://kotlin.github.io/dataframe/convert.html) method, and the data is filtered to include only 
+rows from the year 2000 onwards using the [`.filter`](https://kotlin.github.io/dataframe/filter.html) method.
+
+## Analyze data in Kotlin Notebook
```

Comment:
The 2 steps of this section are already covered in the previous section. What about A) removing "Step 3 Manipulate and filter data" from the "Retrieve section" and placing it inside "Analyze data". You continue the end-to-end data process with the same data set.